File: car-listing-service/src/main/java/org/homecodecarx/car/listing/service/model/entity/CarListing.java
package org.homecodecarx.car.listing.service.model.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import lombok.*;
import org.homecodecarx.car.listing.service.model.enums.ListingStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class CarListing {
    @Id
    private UUID id;
    private UUID sellerId;
    private String title;
    private String description;
    private BigDecimal price;
    private Integer year;
    private Integer mileage;
    private String fuelType;
    private String gearbox;
    private String category;
    private String brand;
    private String model;
    private String location;
    @Enumerated(EnumType.STRING)
    private ListingStatus status;
    private LocalDateTime createdAt;
}

File: carx-client/carx-frontend/src/app/services/car-listing.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";

export interface CreateCarListingRequest {
  sellerId: string;
  title: string;
  description: string;
  price: number;
  year: number;
  mileage: number;
  fuelType: string;
  gearbox: string;
  category: string;
  brand: string;
  model: string;
  location: string;
}

export interface CarListingResponse {
  id: string;
  title: string;
  description: string;
  price: number;
  year: number;
  mileage: number;
  brand: string;
  model: string;
  fuelType: string;
  gearbox: string;
  category: string;
  location: string;
  status: string;
  createdAt: string;
}

export interface CarImageResponse {
  id: string;
  imageUrl: string;
  position: number;
  listingId: string;
}

@Injectable({
  providedIn: 'root'
})
export class CarListingService {
  private apiUrl = 'http://localhost:8081/api/car-listing';

  constructor(private http: HttpClient) {}

  createCarListing(request: CreateCarListingRequest): Observable<CarListingResponse> {
    return this.http.post<CarListingResponse>(this.apiUrl, request);
  }

  uploadImage(listingId: string, file: File, position: number) {
  const formData = new FormData();
  formData.append('image', file);
  formData.append('position', position.toString());

  return this.http.post<CarImageResponse>(
    `/api/listings/${listingId}/images`,
    formData
  );
}
}